src/HavenoDaemon.test.ts

```diff
@@ -119,6 +119,77 @@ test("Can get market prices", async () => {
     .toThrow('Currency not found: INVALID_CURRENCY');
 });
 
+test("Can get market depth information", async () => {
+  await waitForUnlockedBalance(BigInt("1000000000000"), alice);
+
+  // get unlocked balance before reserving funds for offer
+  let unlockedBalanceBefore: bigint = BigInt((await alice.getBalances()).getUnlockedBalance());
+
+  // make buy offers
+  let offer1: OfferInfo = await postOffer(alice, "buy", BigInt("150000000000"));
+  await wait(1000);
+  await waitForUnlockedBalance(BigInt("700000000000"), alice);
+  let offer2: OfferInfo = await postOffer(alice, "buy", BigInt("150000000000"));
+  await wait(1000);
+  await waitForUnlockedBalance(BigInt("400000000000"), alice);
+  let offer3: OfferInfo = await postOffer(alice, "buy", BigInt("200000000000"));
+  await wait(1000);
+
+  // // make sell offers
+  await waitForUnlockedBalance(BigInt("2000000000000"), alice);
+  let offer4: OfferInfo = await postOffer(alice, "sell", BigInt("300000000000"));
+  await wait(1000);
+  await waitForUnlockedBalance(BigInt("1400000000000"), alice);
+  let offer5: OfferInfo = await postOffer(alice, "sell", BigInt("300000000000"));
+  await wait(1000);
+  await waitForUnlockedBalance(BigInt("800000000000"), alice);
+  let offer6: OfferInfo = await postOffer(alice, "sell", BigInt("400000000000"));
+  await wait(1000);
+
+  // get market depth
+  let marketDepthReply: MarketDepthReply = await alice.getMarketDepth("eth");
+  
+  await alice.removeOffer(offer1.getId());
+  await alice.removeOffer(offer2.getId());
+  await alice.removeOffer(offer3.getId());
+  await alice.removeOffer(offer4.getId());
+  await alice.removeOffer(offer5.getId());
+  await alice.removeOffer(offer6.getId());
+
+  // test prices and depth arrays match in length
+  let buyPrices = marketDepthReply.getBuyPricesList();
+  let buyDepth = marketDepthReply.getBuyDepthList().map((depth) => {return BigInt(depth)});
+  expect(buyPrices.length == buyDepth.length);
+
+  let sellPrices = marketDepthReply.getSellPricesList();
+  let sellDepth = marketDepthReply.getSellDepthList().map((depth) => {return BigInt(depth)});
+  expect(sellPrices.length == sellDepth.length);
+
+  // test that prices are positive numbers
+  for (let i in buyPrices) {
+    expect(buyPrices[i]).toBeGreaterThanOrEqual(0);
+  }
+  for (let i in sellPrices) {
+    expect(sellPrices[i]).toBeGreaterThanOrEqual(0);
+  }
+
+  // test that depths and prices are accurate
+  expect(buyPrices[0]).toEqual(1500);
+  expect(buyPrices[1]).toEqual(2000);
+  expect(buyDepth[0]).toEqual(BigInt(2));
+  expect(buyDepth[1]).toEqual(BigInt(1));
+
+  expect(sellPrices[0]).toEqual(3000);
+  expect(sellPrices[1]).toEqual(4000);
+  expect(sellDepth[0]).toEqual(BigInt(2));
+  expect(sellDepth[1]).toEqual(BigInt(1));
+
+  //  // test invalid currency
+  // await expect(async () => {await alice.getMarketDepth("INVALID_CURRENCY")})
+  //   .rejects
+  //   .toThrow('Currency not found: INVALID_CURRENCY');
+});
```

Comment:
Wasn't able to make this work.  Do you think it is necessary?

Reply:
Yes it's necessary to throw if an invalid currency is given, similar to the price call:

```
  // test invalid currency
  await expect(async () => {await alice.getPrice("INVALID_CURRENCY")})
    .rejects
    .toThrow('Currency not found: INVALID_CURRENCY');
```

Reply:
I was having an issue determining what is a valid currency.  Tried a few methods:

1) I looked at the method previously used in CorePriceService.java:66, and this seemed to not apply to the situation of market depth in my mind.  `var marketPrice = priceFeedService.requestAllPrices().get(currencyCode);` checks if a market price exists for a given code, but not necessarily ruling out a valid currency.   can see a situation where the currency is valid but there happens to be no prices available.  Correct me if I'm wrong here please.

2) I looked into the isFiatCurrency and isCryptoCurrency boolean functions.  And on first glance you'd think you could just say `!isFiatCurrency && !isCryptoCurrency` to check for invalid currency, but isCryptoCurrency returns true if the code is not found in Fiat, so there could never be a false value using that construction.